### ISSUE TITLE
Feedback on SnapManual.pdf and related

### ISSUE THREAD

**Issue by gherson** [jmoenig/Snap--Build-Your-Own-Blocks#756]
Mention
* to summon the long input name dialog, for any part of a custom block name, edit the custom block, then click the word of interest in the prototype. p27
* to summon the "Change block" dialog, edit the custom block, then click the prototype outside the block name. 
* also mention, or feel free not to...
 * the useful 'clear' block of the Pen palette.
 * code is easy to delete by dropping it off on to palette.  

Other suggestions (unordered):
* Enable auto-save of project.
* Copy a block's comments when the comments' attached block is copied.
* Add an exponent operator block.
* Enable variable rename. 
* Have the Context menu's delete of a C-command delete only it, and monolithically detach the blocks inside it, rather than deleting them.  
* Besides Undrop, add a general Undo.
* Enable adding the Help's code examples to the Scripting area.
* Add Help for all (primitive) blocks.

Thanks.  Apologies for any ignorance/ingratitude in the above.

**Comment by brianharvey** [comment]
Thanks for the manual suggestions.  I'm way behind on it, and these will get me started.  :-)  About the other stuff, there's an auto-save project in the works; the help system needs to be redone altogether (those help files are pixels, not manipulable text); everyone wants Undo but it's hard; the others we can look at for 4.1 (progress has been made on it...).

**Comment by jmoenig** [comment]
re: Enable variable rename

you know that you *can* rename a variable reporter via its context menu, and it even lets you select from a list of variables reachable in its current context:

![rename var](https://cloud.githubusercontent.com/assets/2596285/6948110/d9ea7614-d8ab-11e4-9ae3-26d395d956b1.png)

And you can also simply select another - reachable - variable in the drop-down of every variable setter. But that's not what you're suggesting, right? You want bulk-rename. But that's problematic because of nested scopes...

**Comment by jmoenig** [comment]
re: Copy a block's comments when the comments' attached block is copied.

Oh, that *should* actually be the case already. Doesn't it work for you?

**Comment by jmoenig** [comment]
re: Have the Context menu's delete of a C-command delete only it, and monolithically detach the blocks inside it, rather than deleting them.

Same here, this is the intended behavior already. Is is not working as you would expect it to?

**Comment by cycomachead** [comment]
> Same here, this is the intended behavior already. Is is not working as you would expect it to?

I don't think this is the case. Like deleting a for loop deletes the stuff inside the for-loop, unless Im misunderstanding the problem?

**Comment by gherson** [comment]
By "Enable variable rename," i meant "Enable global variable rename."

re: "Copy a block's comments when the comments' attached block is copied," I lost comments on a script at some point.  I assumed that point was a copy, but I can't reproduce the issue.  Good to know comment-copy works.

re: "Have the Context menu's delete of a C-command delete only it, and monolithically detach the blocks inside it, rather than deleting them," @cycomachead is correct: when I put some blocks in an IF block, and delete the IF, the blocks inside disappear as well.
